plugins/flytekit-kf-pytorch/flytekitplugins/kfpytorch/task.py

```diff
@@ -455,6 +465,8 @@ def get_custom(self, settings: SerializationSettings) -> Optional[Dict[str, Any]
             )
             job = pytorch_task.DistributedPyTorchTrainingTask(
                 worker_replicas=pytorch_task.DistributedPyTorchTrainingReplicaSpec(
+                    common=plugins_common.CommonReplicaSpec(replicas=self.max_nodes),
+                    # The following fields are deprecated. They are kept for backwards compatibility.
                     replicas=self.max_nodes,
```

Comment:
Will the `replica` arg stay duplicated here?

Reply:
What do you mean? To ensure backward-compatibility, both `replicas` and `common.replicas` need to be sent to the backend.

Reply:
In the respective backend plugin we already distinguish between `taskTemplate.TaskTypeVersion == 0/1`, see [here](https://github.com/flyteorg/flyte/blob/fe879f236d0a4a5d2f6e62c2fdb6b1f21f94cdd6/flyteplugins/go/tasks/plugins/k8s/kfoperators/pytorch/pytorch.go#L60).

I wonder whether removing the duplication in the proto definitions is worth having to check for backwards compatibility in flytekit and flytepluggins. While it might have been better to share the replica spec from the beginning, maybe it's now better to leave as is?

Happy to be proven wrong in this regard!! 🙏 

@pingsutw fyi

Reply:
@fg91 @pingsutw Have you finished the discussion in the contrib sync? What do you think in the end?

Reply:
@fg91 We have duplicated `replica` here to maintain backward compatibility. If people only upgrade flytekit and don't upgrade flyte backend, it should still work, right?

Reply:
Did you test it? If it's duplicated, it should but let's definitely test it.

My personal opinion is that maybe we should have used a shared replica in the first place but duplicating the replica now feels more cluttered than leaving separate replicas.